File: task-03/Elixir/subtask-04.ex
defmodule StarPattern do
  # Function to generate and print the star pattern
  def print_star_pattern(n) do
    top_half =
      for i <- 1..n//1, rem(i, 2) == 1 do
        stars = String.duplicate("*", i)
        padding = String.duplicate(" ", div(n - i, 2))
        "#{padding}#{stars}\n" # Return the line as a string
      end

    bottom_half =
      for i <- (n - 2)..1//-1, rem(i, 2) == 1 do
        stars = String.duplicate("*", i)
        padding = String.duplicate(" ", div(n - i, 2))
        "#{padding}#{stars}\n" # Return the line as a string
      end

    # Concatenate top and bottom halves
    Enum.join(top_half ++ bottom_half)
    |> String.trim() # Remove any trailing newlines
  end

  # Function to read the input and write the pattern to a file
  def write_pattern_to_file do
    {:ok, content} = File.read("input2.txt")
    n = String.trim(content) |> String.to_integer()
    pattern = print_star_pattern(n)

    # Write the pattern to output2.txt
    File.write("output2.txt", pattern)
  end
end

# Call the function to write the pattern to a file
StarPattern.write_pattern_to_file()

File: task-03/Rust/subtask-04.rs
use std::fs;

fn print_star_pattern(n: usize) -> String {
    let mut pattern = String::new();

    for i in (1..=n).step_by(2) {
        let spaces = (n - i) / 2;
        pattern.push_str(&" ".repeat(spaces)); 
        pattern.push_str(&"*".repeat(i)); 
        pattern.push('\n'); 
    }

    for i in (1..n).rev().step_by(2) {
        let spaces = (n - i) / 2;
        pattern.push_str(&" ".repeat(spaces)); 
        pattern.push_str(&"*".repeat(i)); 
        pattern.push('\n'); 
    }

    pattern
}

fn main() -> std::io::Result<()> {
    let input = fs::read_to_string("input2.txt")?;
    let s: usize = input.trim().parse().expect("Please enter a valid number");

    let pattern = print_star_pattern(s);

    fs::write("output2.txt", pattern)?;

    Ok(())
}
